### PR TITLE
feat: Add questions.yaml for telemetry feature

### PR DESCRIPTION
**Pull request by viccuad**
## Description

Relates to https://github.com/kubewarden/helm-charts/issues/310.

## Test

Needs tests against Kubewarden UI.

## Additional Information

PR built on top of https://github.com/kubewarden/helm-charts/pull/331, because this a draft to iterate on questions.yaml.

### CODE REVIEW

**Review by flavio — APPROVED** [review]

File: charts/kubewarden-controller/questions.yaml [hunk 1]
```diff
@@ -58,3 +58,47 @@ questions:
...
+    default: false
+    required: true
+    label: Enable Tracing
+    description: |
+      Enable tracing collection for all PolicyServers
+      Important: Requires OpenTelemetry CRDs available
+    group: "Telemetry"
+    subquestions:
+      - variable: "telemetry.tracing.jaeger.endpoint"
+        type: string # TODO use yaml here
```
[the remark this answers is not in the document]
Comment (viccuad) [reply charts/kubewarden-controller/questions.yaml:90]: correction, this should just have a proper default string, that's all. resolving.

File: charts/kubewarden-controller/questions.yaml [hunk 2]
```diff
@@ -1,60 +1,104 @@
...
+    required: true
+    label: Enable Tracing
+    description: |
+      Enable tracing collection for all PolicyServers
+      Important: Requires OpenTelemetry CRDs available
+    group: "Telemetry"
+    subquestions:
+      - variable: "telemetry.tracing.jaeger.endpoint"
+        type: string
+        default: my-open-telemetry-collector.jaeger.svc.cluster.local:14250"
```
Comment (jordojordo) [inline charts/kubewarden-controller/questions.yaml:91]: Missing a quotation at the beginning of this value.

**Review by jordojordo — CHANGES_REQUESTED** [review]
Small fix in the questions.yml - The questions seem to be rendering correctly aside from the quotation.
__Questions__
chart

__Yaml__
yaml

Running into an issue when installing (I fixed the quotation manually), not sure if this is related:
pod

File: charts/kubewarden-controller/questions.yaml (hunk 2 above)
Comment (viccuad) [reply charts/kubewarden-controller/questions.yaml]: Woops, thanks! Fixed the quotation.

**Comment by viccuad** [comment]
The chart was not installable as we need to merge:
https://github.com/kubewarden/helm-charts/pull/330
https://github.com/kubewarden/helm-charts/pull/331 (included in this PR)

**Comment by viccuad** [comment]
rebased on top of main

**Review by fabriziosestito — APPROVED** [review]

**Review by jordojordo — APPROVED** [review]

File: charts/kubewarden-controller/questions.yaml [hunk 3]
```diff
@@ -1,60 +1,104 @@
...
+          Name of cert-manager Issuer
+        group: "Certificates"
+        show_if: "tls.source=cert-manager"
+  - variable: "telemetry.metrics.enabled"
+    type: boolean
+    default: false
+    required: true
+    label: Enable Metrics
+    description: |
+      Enable metrics collection for all Policy Servers and the Kubewarden Controller
```
Comment (viccuad) [inline charts/kubewarden-controller/questions.yaml:67]:
```suggestion
      Enable metrics collection for all Policy Servers and the Kubewarden Controller.
```

The multiline description string gets rendered as single line. Add dots for readability.

File: charts/kubewarden-controller/questions.yaml [hunk 4]
```diff
@@ -1,60 +1,104 @@
...
+          Port of the Prometheus exporter and PolicyServer metrics service
+        group: "Telemetry"
+        show_if: "telemetry.metrics.enabled=true"
+  - variable: "telemetry.tracing.enabled"
+    type: boolean
+    default: false
+    required: true
+    label: Enable Tracing
+    description: |
+      Enable tracing collection for all PolicyServers
```
Comment (viccuad) [inline charts/kubewarden-controller/questions.yaml:85]:
```suggestion
      Enable tracing collection for all PolicyServers.
```

The multiline description string gets rendered as single line. Add dots for readability.

**Review by flavio — APPROVED** [review]

**Comment by viccuad** [comment]
After consideration, @kravciak raised that the e2e tests are possibly failing because they do a helm upgrade --devel, and there's no rc yet for this changes.

Merging to proceed with an RC.

**Pull request merged**